mesheryctl/pkg/utils/scripts.go

```diff
@@ -6,118 +6,250 @@ import (
 	"os/exec"
 )
 
-// TODO@Aisuko this function is so hard to read and maintain, please refactor it
-func GenerateConfigGKE(configPath, SAName, namespc string) error {
-	script := fmt.Sprintf(`
-	set -e
-	set -o pipefail
-
-	KUBECFG_FILE_NAME="%s"
-	SERVICE_ACCOUNT_NAME=%s
-	NAMESPACE="%s"
-	TARGET_FOLDER=$(dirname ${KUBECFG_FILE_NAME})
-
-	command_exists() {
-		command -v $1 >/dev/null 2>&1
+// GKEConfig holds configuration for GKE script generation
+
+type GKEConfig struct {
+	ConfigPath string
+	SAName     string
+	Namespace  string
+}
+
+// GenerateConfigGKE generates kubernetes config for GKE
+func GenerateConfigGKE(configPath, SAName, namespace string) error {
```

Comment:
```suggestion
func GenerateConfigGKE(configPath, serviceAccountName, namespace string) error {
```

Reply:
@lekaf974 ,i guess in the internal where is this function is being used they are passing it as SAName 
<img width="571" alt="Screenshot 2024-11-21 at 8 16 59 PM" src="https://github.com/user-attachments/assets/030981c7-2dd9-4c35-9f61-98ed2d926ac3">

Reply:
Ok get it back `SAName` if it is already used like that in the code